### ISSUE TITLE
[go_router] Update previous page on pop

### ISSUE THREAD

**Issue by jmysliv** [flutter/flutter#100969]
## Use case

I've noticed that currently is not possible to pass data with `context.pop` to somehow update/refresh the previous page. I know something similar could be achieved using `context.goNamed` but using it will clear the whole stack, and recreate it from the path, which is not always something desirable.

## Proposal

The solution could be to update the previous page with the new `queryParams`, `data` etc.
The API could look like this:
```dart
 void popAndUpdate({
    Map<String, String> params = const <String, String>{},
    Map<String, String> queryParams = const <String, String>{},
    Object? extra,
  }) =>
     // TODO
```

Such a method could pop the current page and updates the previous one with passed arguments, so it could be refreshed.

**Comment by darshankawar** [comment]
@jmysliv 
Can you check if your use case is similar to the one discussed here ? https://github.com/flutter/flutter/issues/100468

**Comment by jmysliv** [comment]
@darshankawar 
Some use cases are similar, but overall it's something different. I want to pass data on pop and the issue you've mentioned is about general behavior on poping with `go()` and `push()`, without passing data.

**Comment by johnpryan** [comment]
I think this is a duplicate of https://github.com/flutter/flutter/issues/99663, please comment if this is not correct.

**Comment by jmysliv** [comment]
@johnpryan it's not duplicate. The issue you are mentioning is about making `push` and `go` a Future that can return value. I suggest adding an extra `pop` method that will allow changing such data as `querParams` or `extra`. This won't require any changes with `push` and `go`.

**Comment by NazarenoCavazzon** [comment]
I created a PR that fixes this https://github.com/flutter/packages/pull/2416

**Comment by jaumard** [comment]
until we get a fix, is there a workaround to update the previous page somehow? Cause I have that use case a lot when building admin pages (list, details/form) and need to update the list once update is done.

**Comment by tot14883** [comment]
I did like below: 
```
   ShellRoute(
          navigatorKey: _shellNavigatorKey,
          builder: (context, state, child) {
            return MainScreen(child: child);
          },
          routes: [
            GoRoute(
              path: '/$dashboardRouteName',
              name: dashboardRouteName,
              pageBuilder: (context, state) => CustomPageRouteBuilder.route(
                key: UniqueKey(), // use this
                child: const DashboardScreen(),
              ),
              routes: [
                GoRoute(
                  path: leaveRequestRouteName,
                  name: '$dashboardRouteName/$leaveRequestRouteName',
                  pageBuilder: (context, state) => CustomPageRouteBuilder.route(
                    key: state.pageKey,
                    child: const LeaveRequestScreen(),
                  ),
                ),
```

**Comment by NazarenoCavazzon** [comment]
I made a package until implementation. They're still defining architecture, so no expectations for near updates.

https://pub.dev/packages/go_router_flow

**Comment by AceChen1** [comment]
@chunhtai Hi, i think pop with value is very common, can merge the code which @NazarenoCavazzon  or other developer's PR ? we are doing the migration with go router from navigator 1.0, but we found go router don't support pop with value, but us project more and more place is already use that, can give some any advice? if continue use navigator 1.0's api, then go router is far away from us...

**Comment by NazarenoCavazzon** [comment]
I made a package for that in pub that I try to keep updated called go_router_flow, you can use it until they add the feature. We're using it in prod in my company.

**Comment by AceChen1** [comment]
> I made a package for that in pub that I try to keep updated called go_router_flow, you can use it until they add the feature. We're using it in prod in my company.

May i know will you keep update the plugin with the latest lib version? looks like go router's version release times is very high frequency.

**Comment by NazarenoCavazzon** [comment]
Yep, I recently updated to the last version of go_router but didn't make the release to pub. It also contains all the corresponding tests.

**Comment by AceChen1** [comment]
> Yep, I recently updated to the last version of go_router but didn't make the release to pub. It also contains all the corresponding tests.

May i know when you plan to release the last version

**Comment by NazarenoCavazzon** [comment]
@AceChen1 I was on a travel in a different city, I made a release just now, updating to 5.2.0, all test passed.

**Comment by AceChen1** [comment]
> @AceChen1 I was on a travel in a different city, I made a release just now, updating to 5.2.0, all test passed.

Sorry to disturb your trip, thank you for your release, and wish you a safe and happy trip

**Comment by NazarenoCavazzon** [comment]
@AceChen1 It seems that while updating to the new version I left a couple things in the road and was not working correctly. I made a new release a few minutes ago that works as intended.

**Comment by AceChen1** [comment]
> @AceChen1 It seems that while updating to the new version I left a couple things in the road and was not working correctly. I made a new release a few minutes ago that works as intended.

Get it. Thanks your update.

**Comment by AceChen1** [comment]
@NazarenoCavazzon  Hi, looks like 5.2.1 had been released...

**Comment by booooohdan** [comment]
No updates?
I just need to call setState when pop. Even the Provider's notifyListener doesn't help
